### PR TITLE
Unify DB tables/columns

### LINKED ISSUE #3449: Refactor: unify database columns
### Clear and concise description of the problem

There are examples of entities:

```go
type Pipeline struct {
	ID                  int64                   `json:"id"                      xorm:"pk autoincr 'pipeline_id'"`
	Status              StatusValue             `json:"status"                  xorm:"INDEX 'pipeline_status'"`
	ChangedFiles        []string                `json:"changed_files,omitempty" xorm:"LONGTEXT 'changed_files'"`
	PullRequestLabels   []string                `json:"pr_labels,omitempty"     xorm:"json 'pr_labels'"`
```

```go
type Step struct {
	ID         int64       `json:"id"                   xorm:"pk autoincr 'step_id'"`
	Type       StepType    `json:"type,omitempty"       xorm:"step_type"`
	Name       string      `json:"name"                 xorm:"step_name"`
	State      StatusValue `json:"state"                xorm:"step_state"`
```

```go
type LogEntry struct {
	ID      int64        `json:"id"       xorm:"pk autoincr 'id'"`
	StepID  int64        `json:"step_id"  xorm:"INDEX 'step_id'"`
	Created int64        `json:"-"        xorm:"created"`
```

### Suggested solution

Get rid of useless prefixes like `pipeline_` in `pipelines` table, `step_` in `steps` table and so on. Exception is link to other entities like `step_id` in `log_entries` table.

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Checked that the feature isn't part of the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't already an [issue](https://github.com/woodpecker-ci/woodpecker/issues) that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by qwerty287**
Closes #3449

### CODE REVIEW

File: server/store/datastore/migration/031_unify_columns_tables.go [hunk 1]
```diff
@@ -0,0 +1,642 @@
...
+		if err := renameColumn(sess, "tasks", "task_labels", "labels"); err != nil {
+			return err
+		}
+		if err := renameColumn(sess, "tasks", "task_dependencies", "dependencies"); err != nil {
+			return err
+		}
+		if err := renameColumn(sess, "tasks", "task_run_on", "run_on"); err != nil {
+			return err
+		}
+		if err := renameColumn(sess, "tasks", "task_dep_status", "dep_status"); err != nil {
```
Comment (anbraten) [inline server/store/datastore/migration/031_unify_columns_tables.go:566]: Maybe `dependencies_status`?

File: server/store/datastore/migration/031_unify_columns_tables.go [hunk 2]
```diff
@@ -0,0 +1,642 @@
...
+		}
+
+		// Workflow
+		if err := renameColumn(sess, "workflows", "workflow_id", "id"); err != nil {
+			return err
+		}
+		if err := renameColumn(sess, "workflows", "workflow_pipeline_id", "pipeline_id"); err != nil {
+			return err
+		}
+		if err := renameColumn(sess, "workflows", "workflow_pid", "pid"); err != nil {
```
Comment (anbraten) [inline server/store/datastore/migration/031_unify_columns_tables.go:609]: `parent_id`?

File: server/model/pipeline.go [hunk 3]
```diff
@@ -20,50 +20,50 @@ import (
...
-	Created             int64                  `json:"created_at"              xorm:"pipeline_created"`
+	ID                  int64                  `json:"id"                      xorm:"pk autoincr 'id'"`
+	RepoID              int64                  `json:"-"                       xorm:"UNIQUE(s) INDEX 'repo_id'"`
+	Number              int64                  `json:"number"                  xorm:"UNIQUE(s) 'number'"`
+	Author              string                 `json:"author"                  xorm:"INDEX 'author'"`
+	Parent              int64                  `json:"parent"                  xorm:"parent"`
+	Event               WebhookEvent           `json:"event"                   xorm:"event"`
+	Status              StatusValue            `json:"status"                  xorm:"INDEX 'status'"`
+	Errors              []*types.PipelineError `json:"errors"                  xorm:"json 'errors'"`
+	Created             int64                  `json:"created_at"              xorm:"created NOT NULL DEFAULT 0 'created'"`
```
Comment (anbraten) [inline server/model/pipeline.go:31]:
```suggestion
	Created             int64                  `json:"created_at"              xorm:"created_at NOT NULL DEFAULT 0 'created_at'"`
```

Would use `_at` suffix for all the timestamps to be consistent with json.

Why is created_at twice in it? Same for `updated_at`. Does it enable some xorm timestamp updating routine?

File: server/store/datastore/migration/031_unify_columns_tables.go (hunk 2 above)
Comment (anbraten) [reply server/store/datastore/migration/031_unify_columns_tables.go:599]: You are right `ppid` is the parent. Not sure if `process id` /  `number` would be a real improvement. We can leave it for now.

File: server/model/cron.go [hunk 4]
```diff
@@ -21,14 +21,14 @@ import (
...
 type Cron struct {
-	ID        int64  `json:"id"                  xorm:"pk autoincr"`
-	Name      string `json:"name"                xorm:"UNIQUE(s) INDEX"`
+	ID        int64  `json:"id"                  xorm:"pk autoincr 'id'"`
+	Name      string `json:"name"                xorm:"name UNIQUE(s) INDEX"`
 	RepoID    int64  `json:"repo_id"             xorm:"repo_id UNIQUE(s) INDEX"`
 	CreatorID int64  `json:"creator_id"          xorm:"creator_id INDEX"`
-	NextExec  int64  `json:"next_exec"`
-	Schedule  string `json:"schedule"            xorm:"NOT NULL"` //	@weekly,	3min, ...
+	NextExec  int64  `json:"next_exec" xorm:"next_exec"`
```
Comment (6543) [inline server/model/cron.go:28]: fmt

File: server/model/feed.go [hunk 5]
```diff
@@ -17,21 +17,21 @@ package model
...
-	Commit   string `json:"commit,omitempty"        xorm:"feed_pipeline_commit"`
-	Branch   string `json:"branch,omitempty"        xorm:"feed_pipeline_branch"`
-	Ref      string `json:"ref,omitempty"           xorm:"feed_pipeline_ref"`
-	Refspec  string `json:"refspec,omitempty"       xorm:"feed_pipeline_refspec"`
-	Title    string `json:"title,omitempty"         xorm:"feed_pipeline_title"`
-	Message  string `json:"message,omitempty"       xorm:"feed_pipeline_message"`
-	Author   string `json:"author,omitempty"        xorm:"feed_pipeline_author"`
-	Avatar   string `json:"author_avatar,omitempty" xorm:"feed_pipeline_avatar"`
-	Email    string `json:"author_email,omitempty"  xorm:"feed_pipeline_email"`
+	RepoID   int64  `json:"repo_id"                 xorm:"repo_id"`
```
Comment (6543) [inline server/model/feed.go:20]: be aware dragons lying here ... this is a "virtual table" created by a select statement

File: server/model/pipeline.go (hunk 3 above)
Comment (anbraten) [reply server/model/pipeline.go:31]: I would just rename the xorm columns here?

File: server/store/datastore/migration/031_unify_columns_tables.go [hunk 6]
```diff
@@ -0,0 +1,642 @@
...
+		if err := renameColumn(sess, "crons", "i_d", "id"); err != nil {
+			return err
+		}
+
+		// Forge
+		if err := renameTable(sess, "forge", "forges"); err != nil {
+			return err
+		}
+
+		// Perm
```
[the remark this answers is not in the document]
Comment (anbraten) [reply server/store/datastore/migration/031_unify_columns_tables.go:279]: Its a weird abbreviation IMO, but we leave it for now.

File: server/model/feed.go (hunk 5 above)
Comment (anbraten) [reply server/model/feed.go:20]: I've compared most parts side-by-side and could find anything in the first place

File: server/model/pipeline.go (hunk 3 above)
Comment (anbraten) [reply server/model/pipeline.go:31]: Looks like we have evenly used `created` and `created_at` feel free to pick your preference for xorm and we will migrate the json property in 3.0

File: server/model/feed.go [hunk 7]
```diff
@@ -17,21 +17,21 @@ package model
...
+	ID       int64  `json:"id,omitempty"            xorm:"pipeline_id"`
+	Number   int64  `json:"number,omitempty"        xorm:"pipeline_number"`
+	Event    string `json:"event,omitempty"         xorm:"pipeline_event"`
+	Status   string `json:"status,omitempty"        xorm:"pipeline_status"`
+	Created  int64  `json:"created_at,omitempty"    xorm:"pipeline_created"` // TODO change JSON field to "created" in 3.0
+	Started  int64  `json:"started_at,omitempty"    xorm:"pipeline_started"`
+	Finished int64  `json:"finished_at,omitempty"   xorm:"pipeline_finished"`
```
Comment (anbraten) [inline server/model/feed.go:27]:
```suggestion
	Started  int64  `json:"started_at,omitempty"    xorm:"pipeline_started"` // TODO change JSON field to "started" in 3.0
	Finished int64  `json:"finished_at,omitempty"   xorm:"pipeline_finished"` // TODO change JSON field to "finished" in 3.0
```

File: server/model/pipeline.go [hunk 8]
```diff
@@ -20,50 +20,50 @@ import (
...
+	Number              int64                  `json:"number"                  xorm:"UNIQUE(s) 'number'"`
+	Author              string                 `json:"author"                  xorm:"INDEX 'author'"`
+	Parent              int64                  `json:"parent"                  xorm:"parent"`
+	Event               WebhookEvent           `json:"event"                   xorm:"event"`
+	Status              StatusValue            `json:"status"                  xorm:"INDEX 'status'"`
+	Errors              []*types.PipelineError `json:"errors"                  xorm:"json 'errors'"`
+	Created             int64                  `json:"created_at"              xorm:"created NOT NULL DEFAULT 0 'created'"` // TODO change JSON field to "created" in 3.0
+	Updated             int64                  `json:"updated_at"              xorm:"updated NOT NULL DEFAULT 0 'updated'"` // TODO change JSON field to "updated" in 3.0
+	Started             int64                  `json:"started_at"              xorm:"started"`
+	Finished            int64                  `json:"finished_at"             xorm:"finished"`
```
Comment (anbraten) [inline server/model/pipeline.go:34]:
```suggestion
	Started             int64                  `json:"started_at"              xorm:"started"` // TODO change JSON field to "started" in 3.0
	Finished            int64                  `json:"finished_at"             xorm:"finished"` // TODO change JSON field to "finished" in 3.0
```

File: server/model/pipeline.go [hunk 9]
```diff
@@ -20,50 +20,50 @@ import (
...
+	Refspec             string                 `json:"refspec"                 xorm:"refspec"`
+	Title               string                 `json:"title"                   xorm:"title"`
+	Message             string                 `json:"message"                 xorm:"TEXT 'message'"`
+	Timestamp           int64                  `json:"timestamp"               xorm:"timestamp"`
+	Sender              string                 `json:"sender"                  xorm:"sender"` // uses reported user for webhooks and name of cron for cron pipelines
+	Avatar              string                 `json:"author_avatar"           xorm:"avatar"`
+	Email               string                 `json:"author_email"            xorm:"email"`
+	ForgeURL            string                 `json:"forge_url"               xorm:"forge_url"`
+	Reviewer            string                 `json:"reviewed_by"             xorm:"reviewer"`
+	Reviewed            int64                  `json:"reviewed_at"             xorm:"reviewed"`
```
Comment (anbraten) [inline server/model/pipeline.go:49]:
```suggestion
	Reviewed            int64                  `json:"reviewed_at"             xorm:"reviewed"` // TODO change JSON field to "reviewed" in 3.0
```

**Review by anbraten — APPROVED** [review]

File: server/store/datastore/feed.go [hunk 10]
```diff
@@ -20,30 +20,30 @@ import (
...
-pipelines.pipeline_avatar as feed_pipeline_avatar`
+var feedItemSelect = `repos.id as repo_id,
+pipelines.id as pipeline_id,
+pipelines.number as pipeline_number,
+pipelines.event as pipeline_event,
+pipelines.status as pipeline_status,
+pipelines.created as pipeline_created,
+pipelines.started as pipeline_started,
+pipelines.finished as pipeline_finished,
+'pipelines.commit' as pipeline_commit,
```
[the remark this answers is not in the document]
Comment (xoxys) [reply server/store/datastore/feed.go:31]: @qwerty287 @anbraten do you remember the exact error? This is the root cause for https://github.com/woodpecker-ci/woodpecker/issues/4537

Removing the quoting fixes the linked issue, and I can also compile and run the server without any problem.

[the remark this answers is not in the document]
Comment (xoxys) [reply server/store/datastore/feed.go:31]: Ok got it :+1:

**Pull request merged**